### PR TITLE
Extract time query parameter in splunkhecreceiver

### PR DESCRIPTION
**Pull request by splunkericl**
**Description:** 
This change adds a new feature in splunk hec receiver allowing users to specify time query parameter. This is to put parity between splunk hec receiver and splunk HEC raw endpoint

Note: the validation response is slightly different than what splunk returns. if invalid input is provided for time:
- splunk returns 400 response with this error message:
```
{
    "text": "Error in handling indexed fields",
    "code": 15,
    "invalid-event-number": 0
}
```
- however this doesn't make sense for splunk hec receiver as it is not indexing anything. Instead, splunk hec receiver will return:
```
{"text":"Invalid data format","code":6}
```

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27006

**Testing:** Update unit test to verify both paths

**Documentation:** N/A

### CODE REVIEW

**Comment by timannguyen** [comment]
LGTM

File: .chloggen/splunkhecreceiver-add-time-param.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: [api]
```
Comment (atoulme) [inline .chloggen/splunkhecreceiver-add-time-param.yaml:27]:
```suggestion
change_logs: [user]
```
This is a user changelog-worthy entry imo, as it changes the input options.

**Review by atoulme — APPROVED** [review]

**Comment by splunkericl** [comment]
@atoulme can we merge this change in?

**Comment by atoulme** [comment]
I'm just an approver, I can't merge anything. A maintainer will be in touch.

**Pull request merged**